common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -296,10 +301,15 @@ protected void serviceInit(Configuration externalConf) throws Exception {
           DEFAULT_SPARK_SHUFFLE_SERVICE_METRICS_NAME);
       YarnShuffleServiceMetrics serviceMetrics =
           new YarnShuffleServiceMetrics(metricsNamespace, blockHandler.getAllMetrics());
+      YarnShuffleServiceMetrics mergeManagerMetrics =
+          new YarnShuffleServiceMetrics("mergeManagerMetrics", blockPushResolver.getMetrics());
```

Comment:
We should not be directly creating `RemoteBlockPushResolver` within service init - it is done via `newMergedShuffleFileManagerInstance`, so that it returns the right instance configured by admin (by default disabled).

We should be adding a `getMetrics` to `MergedShuffleFileManager` to retrieve the metrics we want to register.

Reply:
Thanks, the UTs passed after the fix.